tools/progen_compile.py

```diff
@@ -88,16 +90,26 @@ def build_projects(file, projects, tool, build=True, clean=False, ignore_errors=
                     nargs='*', type=str, default=[])
 parser.add_argument('--release', dest='release', action='store_true', help='Create a release with the yaml version file')
 parser.add_argument('--release-folder', type=str, default='firmware', help='Directory to create and place files in')
-parser.add_argument('--toolchain', type=str, default='GCC', help='Toolchain ("GCC" or "ARM", default: %(default)s',
-                    choices=['GCC', 'ARM'])
+parser.add_argument('--toolchain', type=str, default='GCC', help='Toolchain ("GCC", "ARMCC", "ARMCLANG", default: %(default)s',
+                    choices=['GCC', 'ARMCC', "ARMCLANG"])
```

Comment:
Check the generated help. I think it already lists available choices when you provide a `choices` keyword arg.

In any case, the close parenthesis is missing in the help string.

(I don't like using `choices` because it uses case-sensitive matching.)